### ISSUE TITLE
PostgreSQL DeadLock

### ISSUE THREAD

**Issue by djpremier** [nextcloud/server#9961]
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'UPDATE "oc_filecache" SET "mtime" = GREATEST("mtime", ?), "etag" = ? WHERE ("storage" = ?) AND ("path_hash" IN ('d41d8cd98f00b204e9800998ecf8427e', 'ba6faea51fc5e9a600aff43aa64fb140', 'f4d6b76891f39a882516952e6dc844d5', '992738f64bef805fe8d514c43e6e2f8d'))' with params [1529583888, "5b2b9910bd255", 1]: SQLSTATE[40P01]: Deadlock detected: 7 ERROR: deadlock detected DETAIL: Process 16610 waits for ShareLock on transaction 54947; blocked by process 16616. Process 16616 waits for ShareLock on transaction 54951; blocked by process 16610. HINT: See server log for query details. CONTEXT: while rechecking updated tuple (20,20) in relation "oc_filecache"

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'UPDATE "oc_filecache" SET "mtime" = GREATEST("mtime", ?), "etag" = ? WHERE ("storage" = ?) AND ("path_hash" IN ('d41d8cd98f00b204e9800998ecf8427e', 'ba6faea51fc5e9a600aff43aa64fb140', 'f4d6b76891f39a882516952e6dc844d5', '5c60ca5dac61c6b173aa6da2b2e79709'))' with params [1529583888, "5b2b9910af20b", 1]: SQLSTATE[40P01]: Deadlock detected: 7 ERROR: deadlock detected DETAIL: Process 16613 waits for ShareLock on transaction 54948; blocked by process 16609. Process 16609 waits for ShareLock on transaction 54945; blocked by process 16613. HINT: See server log for query details. CONTEXT: while rechecking updated tuple (20,17) in relation "oc_filecache"

**Comment by MorrisJobke** [comment]
Seems to be related to #6104. Are you on 13.0.4? And please fill out the issue template from https://raw.githubusercontent.com/nextcloud/server/master/issue_template.md

**Comment by torrentkino** [comment]
sudo cat /var/log/postgresql/postgresql-9.6-main.log.1 | egrep "nextcloud.*deadlock detected" | wc -l
3474

Debian/Stretch / postgresql-9.6 / Nextcloud 14.0.3 / Only one user

**Comment by danielkesselberg** [comment]
Hey @torrentkino, 

if you are running into this issue please open a new ticket. Could you try Nextcloud 14.0.4? They were many fixes https://github.com/nextcloud/server/pull/12433.

**Comment by torrentkino** [comment]
@danielkesselberg Will do. Thank you.
